homework2/task1.c

```diff
@@ -0,0 +1,98 @@
+#include <stdio.h>
+#include <stdbool.h>
+
+
+int recursiveFibonacciNumber(int number)
+{
+    if (number < 3)
+    {
+        return 1;
+    }
+    else
+    {
+        return recursiveFibonacciNumber(number - 1) + recursiveFibonacciNumber(number - 2);
+    }
+}
+//при n >= 39 уже очень заметно
+
+
+int iterativeFibonacciNumber(int number)
+{
+    int queueOfTwoNumbers[2] = { 1, 1 };
+    int currentFibonacciNumber = 1;
+    for (int i = 2; i < number; ++i)
+    {
+        currentFibonacciNumber = queueOfTwoNumbers[0] + queueOfTwoNumbers[1];
+        queueOfTwoNumbers[0] = queueOfTwoNumbers[1];
+        queueOfTwoNumbers[1] = currentFibonacciNumber;
+    }
+    return currentFibonacciNumber;
+}
+
+
+bool test1(void)
+{
+    for (int i = 0; i < 35; ++i)
+    {
+        if (recursiveFibonacciNumber(i) != iterativeFibonacciNumber(i))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+
+bool test2(void)
+{
+    if (recursiveFibonacciNumber(36) != 14930352 || iterativeFibonacciNumber(36) != 14930352)
+    {
+        return false;
+    }
+    return true;
```

Comment:
```suggestion
     return recursiveFibonacciNumber(36) == 14930352 && iterativeFibonacciNumber(36) == 14930352
```
Условие условного оператора - уже булево значение.